educe/graph.py

```diff
@@ -502,16 +502,15 @@ def cdu_members(self, cdu, deep=False):
         that are members of (members of ..) this CDU.
         """
 
+        hyperedge = self.edgeform(cdu)
+        members = set(self.links(hyperedge))
+
         if deep:
-            members = set()
-            for m in self.cdu_members(cdu):
-                members.add(m)
+            for m in list(members):
```

Comment:
Now I remember, after it crashes: you can't add elements to a set while you are iterating over it. Hence my copy of `members` into a list. I could have copied it into a new set, though...